File: semana02/dia05/flask_blog_api/routes/user_router.py
from flask import Blueprint, request
from controllers.user_controller import UserController
from flask_jwt_extended import jwt_required


user_router = Blueprint('user', __name__)
controller = UserController()


@user_router.post('/create')
@jwt_required()
def create_user():
    json = request.get_json()
    return controller.create(json)

@user_router.get('/list')
@jwt_required()
def list_users():
    return controller.list()

@user_router.put('/update/<int:id>')
@jwt_required()
def update_user(id):
    json = request.get_json()
    return controller.update(id, json)

@user_router.get('/get/<int:id>')
@jwt_required()
def get_user(id):
    return controller.get_by_id(id)

@user_router.delete('/delete/<int:id>')
@jwt_required()
def delete_user(id):
    return controller.delete(id)

File: semana07/express_prisma/src/routes/user.router.ts
import { Router } from "express";
import * as userController from "../controllers/user.controller";

export const userRouter = Router();

userRouter
  /**
   * @swagger
   * /api/user/create:
   *   post:
   *     summary: Ruta para crear un usuario
   *     description: Crear un usuario
   *     tags: [Usuario]
   *     requestBody:
   *       required: true
   *       content:
   *         application/json:
   *           schema:
   *             $ref: '#/components/schemas/User'
   *     responses:
   *       201:
   *         description: Usuario creado exitosamente
   *         content:
   *           application/json:
   *             schema:
   *               $ref: '#/components/schemas/CreateUserResponse'
   */
  .post("/create", userController.createUser)
  /**
   * @swagger
   * /api/user/list:
   *   get:
   *     summary: Ruta para listar usuarios
   *     description: Listar usuarios
   *     tags: [Usuario]
   *     responses:
   *       200:
   *         description: Lista de usuarios
   *         content:
   *           application/json:
   *             schema:
   *               $ref: '#/components/schemas/UserListResponse'
   */
  .get("/list", userController.listUsers)
  .get("/by-id/:userId", userController.getUserById)
  .put("/update/:userId", userController.updateUser)
  .delete("/delete/:userId", userController.deleteUser);
